### PR TITLE
[MIG] purchase_analytic: Migration to 16.0

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
@qrtl

### CODE REVIEW

**Review by RodrigoBM — APPROVED** [review]
LGFM!

File: purchase_analytic/models/purchase.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        states={"draft": [("readonly", False)]},
+        store=True,
+        help="The analytic distribution related to a purchase order.",
+    )
+    analytic_precision = fields.Integer(
+        store=False,
+        default=lambda self: self.env["decimal.precision"].precision_get(
+            "Percentage Analytic"
+        ),
+    )
```
Comment (yostashiro) [inline purchase_analytic/models/purchase.py]:
```suggestion
    _name = "purchase.order"
    _inherit = ["purchase.order", "analytic.mixin"]

    analytic_distribution = fields.Json(
        inverse="_inverse_analytic_distribution",
        readonly=True,
        states={"draft": [("readonly", False)]},
    )
```

File: purchase_analytic/views/purchase_views.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        <field name="name">purchase.order.form (purchase_analytic)</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form" />
+        <field name="arch" type="xml">
+            <field name="partner_ref" position="after">
+                <field
+                    name="analytic_distribution"
+                    widget="analytic_distribution"
+                    groups="analytic.group_analytic_accounting"
+                    options="{'product_field': 'product_id', 'business_domain': 'purchase_order'}"
```
Comment (yostashiro) [inline purchase_analytic/views/purchase_views.xml]:
```suggestion
```
I think these options are of no use here.

**Review by yostashiro — CHANGES_REQUESTED** [review]

**Review by yostashiro — APPROVED** [review]
LGTM. 👍

@AungKoKoLin1997 Please squash your migration commits.

**Review by AaronHForgeFlow — APPROVED** [review]
Functional LG

File: purchase_analytic/views/purchase_views.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        <field name="name">purchase.order.tree (purchase_analytic)</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_view_tree" />
+        <field name="arch" type="xml">
+            <field name="origin" position="before">
+                <field
+                    name="analytic_distribution"
+                    widget="analytic_distribution"
+                    groups="analytic.group_analytic_accounting"
+                />
```
Comment (yostashiro) [inline purchase_analytic/views/purchase_views.xml]:
```suggestion
                <field
                    name="analytic_distribution"
                    widget="analytic_distribution"
                    groups="analytic.group_analytic_accounting"
                    optional="show"
                />
```
Got feedback from a user that they want to be able to hide the column.

**Review by yostashiro — COMMENTED** [review]
@AungKoKoLin1997 Can you please also add the analytic_distribution field in the RFQ tree view (purchase.purchase_order_kpis_tree) in the same manner as in the purchase order tree view.

**Comment by yostashiro** [comment]
@AungKoKoLin1997 Thanks! LGTM.

File: purchase_analytic/models/purchase.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+
+class PurchaseOrder(models.Model):
+    _name = "purchase.order"
+    _inherit = ["purchase.order", "analytic.mixin"]
+
+    analytic_distribution = fields.Json(
+        inverse="_inverse_analytic_distribution",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
```
Comment (yostashiro) [inline purchase_analytic/models/purchase.py]:
```suggestion
        # To be consistent with order_line readonly behavior
        states={"done": [("readonly", True)], "cancel": [("readonly", True)]},
```
@AungKoKoLin1997

**Review by yostashiro — APPROVED** [review]
👍

**Comment by AaronHForgeFlow** [comment]
@AungKoKoLin1997 can you check pre-commit? maybe a rebase is enough...

**Comment by AungKoKoLin1997** [comment]
> can you check pre-commit? maybe a rebase is enough...

@AaronHForgeFlow Rebase can't be solved pre-commit issues at the moment. I will rebase again this PR after [that PR](https://github.com/OCA/account-analytic/pull/535) is already merged.

**Comment by celm1990** [comment]
Hi, I think this module needs a migration script to convert the old project_id field to the new analytic_distribution field. Or is this handled in another way?

Thanks in advance.

**Comment by AungKoKoLin1997** [comment]
> Hi, I think this module needs a migration script to convert the old project_id field to the new analytic_distribution field.

@celm1990 You are right. I missed handling that when I worked on this PR.
I will follow up on this in the coming weeks. Please feel free to work on it if you need it urgently.

**Pull request merged**